### PR TITLE
NOT READY : Implement transformation into rgb/a & cover

### PR DESCRIPTION
**Pull request by JackieJoo**
.

### CODE REVIEW

File: proto/wtools/amid/color/Color.s [hunk 1]
```diff
@@ -1155,6 +1155,424 @@ function paler( rgb, factor )
...
+//
+
+function cmykStructureStrToRgb( src )
+{
+  /* C100/M80/Y0/K35 */
+
+  _.assert( arguments.length === 1, 'Expects single argument' );
+  _.assert( _.strIs( src ) );
+
+  let result = cmykStructureStrToRgba.call( this, src );
```
Comment (Wandalen) [inline proto/wtools/amid/color/Color.s:1259]: bad

File: proto/wtools/amid/color/Color.s [hunk 2]
```diff
@@ -1155,6 +1155,424 @@ function paler( rgb, factor )
...
+
+  if( result )
+  return _.longSlice( result, 0, 3 );
+
+  return null;
+}
+
+//
+
+function cmykStructureStrToRgba( src )
```
Comment (Wandalen) [inline proto/wtools/amid/color/Color.s:1269]: bad name

File: proto/wtools/amid/color/Color.s [hunk 3]
```diff
@@ -1155,6 +1155,424 @@ function paler( rgb, factor )
...
+
+  if( result )
+  return _.longSlice( result, 0, 3 );
+
+  return null;
+}
+
+//
+
+function hwbStrToRgba( src )
```
Comment (Wandalen) [inline proto/wtools/amid/color/Color.s:1303]: bad

File: proto/wtools/amid/color/Color.s [hunk 4]
```diff
@@ -1155,6 +1155,433 @@ function paler( rgb, factor )
...
+function _hslaStrToRgba( src )
+{
+  _.assert( arguments.length === 1, 'Expects single argument' );
+  _.assert( _.strIs( src ) );
+
+  let [ h, s, l, a ] = _.color._formatStringParse( src );
+
+  if
+  (
+    !_.color._validateRangeInclusive( h, 0, 360 )
```
Comment (Wandalen) [inline proto/wtools/amid/color/Color.s:1424]: no

File: proto/wtools/amid/color/Color.s [hunk 5]
```diff
@@ -1305,18 +1732,18 @@ function _rgbWithInt( srcInt )
 // hsl
 // --
 
-function hslToRgb( hsl, result )
+function _hslToRgb( hsl, result )
```
Comment (Wandalen) [inline proto/wtools/amid/color/Color.s:1735]: breaks styles

File: proto/wtools/amid/color/Color.s [hunk 6]
```diff
@@ -1155,6 +1155,433 @@ function paler( rgb, factor )
...
+}
+
+//
+
+function _labStrToRgb( src )
+{
+  _.assert( arguments.length === 1, 'Expects single argument' );
+  _.assert( _.strIs( src ) );
+
+  let result = _.color._labStrToRgba( src );
```
Comment (Wandalen) [inline proto/wtools/amid/color/Color.s:1511]: bad. not optimal

File: proto/wtools/amid/color/Color.s (hunk 4 above)
Comment (JackieJoo) [reply proto/wtools/amid/color/Color.s]: Changed

File: proto/wtools/amid/color/Color.s (hunk 6 above)
Comment (JackieJoo) [reply proto/wtools/amid/color/Color.s:1511]: Alpha channel is not used in the conversion, it's just added if present to the resulting array of rgba values. Running times would be the same for `*ToRgb` and `*ToRgba`, so it's more readable to use `*ToRgba` routines and slice 3 elements from the result to retrieve rgb values.

File: proto/wtools/amid/color/Color.s (hunk 5 above)
Comment (JackieJoo) [reply proto/wtools/amid/color/Color.s]: Revert changes, routine will be extended in later PRs.

File: proto/wtools/amid/color/Color.s [hunk 7]
```diff
@@ -1155,6 +1155,428 @@ function paler( rgb, factor )
...
+      f = 1 - f;
+    }
+
+    n = wh + f * ( v - wh ); // linear interpolation
+
+    let r, g, b;
+
+    switch( i )
+    {
+    case 6 :
```
Comment (Wandalen) [inline proto/wtools/amid/color/Color.s:1288]: break of styles

File: proto/wtools/amid/color/Color.s [hunk 8]
```diff
@@ -1155,6 +1155,428 @@ function paler( rgb, factor )
...
+
+  if( result )
+  return _.longSlice( result, 0, 3 );
+
+  return null;
+}
+
+//
+
+function _cmykStrToRgba( src )
```
Comment (Wandalen) [inline proto/wtools/amid/color/Color.s:1182]: scales should be normalized

File: proto/wtools/amid/color/Color.s (hunk 6 above)
Comment (Wandalen) [reply proto/wtools/amid/color/Color.s:1511]: no

**Pull request closed without merging**